backend/elasticsearch/src/main/java/org/hibernate/search/backend/elasticsearch/types/projection/impl/GeoPointFieldProjectionBuilderFactory.java

```diff
@@ -37,7 +37,7 @@ public boolean isDslCompatibleWith(ElasticsearchFieldProjectionBuilderFactory ob
 		if ( this == obj ) {
 			return true;
 		}
-		if ( !( obj instanceof GeoPointFieldProjectionBuilderFactory ) ) {
```

Comment:
You're right, it's better. While you're at it, maybe you could fix `GeoPointFieldPredicateBuilderFactory#isDslCompatibleWith` (the Lucene one): it just compares the classes, but I find your pattern of comparing `INSTANCE == obj` more foolproof (if we remove the `INSTANCE` constant one day, we'll have to have a look at the `isDslCompatibleWith` method).

Reply:
> (if we remove the INSTANCE constant one day, we'll have to have a look at the isDslCompatibleWith method)

That's exactly why I did it that way :).

Fixed `GeoPointFieldPredicateBuilderFactory`.